AreaCalculator.java

```diff
@@ -2,40 +2,47 @@
 public class AreaCalculator {
 	
 	  public static double calculate(String... params) { 
+	
 		  	if (params.length < 2)
-		  		return 0 ;
-		  	double area = 0 ;
-		  	if (params[0].equals("square") && params.length==3 )
+		  		return -1 ;
```

Comment:
Generally, this is OK - as `-1` is indeed an impossible return value for an area calculation. But that's not correct either. An area of a circle with a negative radius is not -1, its just impossible.
Error return codes are highly important. They need to be "self-describing" and very correct, so its either - a constant you set or use that can't be confused with an actual number, such as `Double.NaN`, or an exception you throw such as `throw new IllegalArgumentException("params.length < 2")`, note the `String` describing the problem - its important for whoever is catching the error.
Whether to use a constant or an exception is a **design choice**. Lets talk about that in chat, but I think here an exception is the right choice.